test/react_test_utils_test.dart

```diff
@@ -229,6 +229,22 @@ void main() {
     expect(results[1]['tagName'], equals('DIV'));
   });
 
+  test('scryRenderedDomComponentsWithId', () {
+    var idName = 'searchableId';
+
+    component = renderIntoDocument(div({}, [
+      div({'id': idName}),
+      div({'id': idName}),
+      span({})
```

Comment:
It wouldn't be a bad idea to add another element with a different ID to verify that not all elements with IDs are being returned.